coresdk/src/backend/gpio_driver.cpp

```diff
@@ -0,0 +1,140 @@
+/***********************************************
+ * XQuestCode || Aditya Parmar
+ * © 2024 Aditya Parmar. All Rights Reserved.
+ * This file is part of the SplashKit Core Library.
+ * Use https://abyz.me.uk/rpi/pigpio/pdif2.html for reference
+ ***********************************************/
+
+#include "gpio_driver.h"
+
+#include <string>
+#include <iostream>
+#include <cstdlib> // Add this line to include the necessary header for the exit() function
+
+#ifdef RASPBERRY_PI
+#include "pigpiod_if2.h"
+#endif
+
+using namespace std;
+
+namespace splashkit_lib
+{
+    int pi;
+
+    // Check if pigpio_init() has been called before any other GPIO functions
+    void check_pi()
+    {
+#ifdef RASPBERRY_PI
+        if (pi < 0)
+        {
+            cout << "pigpio_init() must be called before any other GPIO functions" << endl;
+            exit(1);
```

Comment:
Dont terminate if there is an issue - make this a function that returns a boolean. Then the calling code can have `if(check_pi()) ...`